packages/dev/core/src/Engines/WebGPU/webgpuTextureManager.ts

```diff
@@ -31,135 +31,130 @@ import { InternalTextureSource } from "../../Materials/Textures/internalTexture"
+            vertexOutputs.vTextureSize = vec2f(textureDimensions(img, 0));
         #endif
-        gl_Position = vec4(pos[gl_VertexIndex], 0.0, 1.0);
+        vertexOutputs.position =  vec4f(pos[input.vertexIndex], 0.0, 1.0);
     }
     `;
 
 const invertYPreMultiplyAlphaFragmentSource = `
-    #extension GL_EXT_samplerless_texture_functions : enable
-
-    layout(set = 0, binding = 0) uniform texture2D img;
+    var img: texture_2d<f32>;
 
     #ifdef INVERTY
-        layout(location = 0) in flat ivec2 vTextureSize;
+        varying vTextureSize: vec2f;
     #endif
-    layout(location = 0) out vec4 outColor;
 
-    void main() {
+    @fragment
+    fn main(input: FragmentInputs) -> FragmentOutputs {
     #ifdef INVERTY
-        vec4 color = texelFetch(img, ivec2(gl_FragCoord.x, vTextureSize.y - gl_FragCoord.y), 0);
+        var color: vec4f = textureLoad(img, vec2i(i32(input.position.x), i32(input.vTextureSize.y - input.position.y)), 0);
     #else
-        vec4 color = texelFetch(img, ivec2(gl_FragCoord.xy), 0);
+        var color: vec4f = textureLoad(img, vec2i(input.position.xy), 0);
     #endif
     #ifdef PREMULTIPLYALPHA
-        color.rgb *= color.a;
+        fragmentOutputs.color = vec4f(color.rgb * color.a, color.a);
     #endif
-        outColor = color;
+        fragmentOutputs.color = color;
     }
     `;
 
 const invertYPreMultiplyAlphaWithOfstVertexSource = invertYPreMultiplyAlphaVertexSource;
 
 const invertYPreMultiplyAlphaWithOfstFragmentSource = `
-    #extension GL_EXT_samplerless_texture_functions : enable
-
-    layout(set = 0, binding = 0) uniform texture2D img;
-    layout(set = 0, binding = 1) uniform Params {
-        float ofstX;
-        float ofstY;
-        float width;
-        float height;
-    };
+    var img: texture_2d<f32>;
+    uniform ofstX: f32;
+    uniform ofstY: f32;
+    uniform width: f32;
+    uniform height: f32;
 
     #ifdef INVERTY
-        layout(location = 0) in flat ivec2 vTextureSize;
+        varying vTextureSize: vec2f;
     #endif
-    layout(location = 0) out vec4 outColor;
 
-    void main() {
-        if (gl_FragCoord.x < ofstX || gl_FragCoord.x >= ofstX + width) {
+    @fragment
+    fn main(input: FragmentInputs) -> FragmentOutputs {
+        if (input.position.x < uniforms.ofstX || input.position.x >= uniforms.ofstX + width) {
             discard;
         }
-        if (gl_FragCoord.y < ofstY || gl_FragCoord.y >= ofstY + height) {
+        if (input.position.y < uniforms.ofstY || input.position.y >= uniforms.ofstY + height) {
             discard;
         }
     #ifdef INVERTY
-        vec4 color = texelFetch(img, ivec2(gl_FragCoord.x, ofstY + height - (gl_FragCoord.y - ofstY)), 0);
+        var color: vec4f = textureLoad(img, vec2i(input.position.x, ofstY + height - (input.position.y - uniforms.ofstY)), 0);
     #else
-        vec4 color = texelFetch(img, ivec2(gl_FragCoord.xy), 0);
+        var color: vec4f = textureLoad(img, vec2i(input.position.xy), 0);

```

Comment:
In this block of code, some `uniforms.` and some `i32(...)` casts are missing.

You can use [this PG](http://localhost:1338/#4HG7NY#6) to check this codepath.

Fixed code inside `main`:
```wgsl
        if (input.position.x < uniforms.ofstX || input.position.x >= uniforms.ofstX + uniforms.width) {
            discard;
        }
        if (input.position.y < uniforms.ofstY || input.position.y >= uniforms.ofstY + uniforms.height) {
            discard;
        }
    #ifdef INVERTY
        var color: vec4f = textureLoad(img, vec2i(i32(input.position.x), i32(uniforms.ofstY + uniforms.height - (input.position.y - uniforms.ofstY))), 0);
    #else
        var color: vec4f = textureLoad(img, vec2i(input.position.xy), 0);
    #endif
    #ifdef PREMULTIPLYALPHA
        color = vec4f(color.rgb * color.a, color.a);
    #endif
        fragmentOutputs.color = color;
```